src/Sylius/Component/Core/spec/Model/OrderItemSpec.php

```diff
@@ -97,24 +99,29 @@ function it_returns_unit_price_as_discounted_unit_price_if_there_are_no_units():
         $this->getDiscountedUnitPrice()->shouldReturn(10000);
     }
 
-    function it_returns_subtotal_which_consist_of_discounted_unit_price_multiplied_by_quantity(
-        OrderItemUnitInterface $firstUnit,
-        OrderItemUnitInterface $secondUnit
-    ): void {
+    function its_subtotal_consist_of_sum_of_units_discounted_price(): void
+    {
         $this->setUnitPrice(10000);
 
-        $firstUnit->getOrderItem()->willReturn($this);
-        $firstUnit->getTotal()->willReturn(9000);
-        $firstUnit->getAdjustmentsTotal(AdjustmentInterface::ORDER_UNIT_PROMOTION_ADJUSTMENT)->willReturn(-500);
-
-        $secondUnit->getOrderItem()->willReturn($this);
-        $secondUnit->getTotal()->willReturn(9000);
-        $secondUnit->getAdjustmentsTotal(AdjustmentInterface::ORDER_UNIT_PROMOTION_ADJUSTMENT)->willReturn(-500);
-
-        $this->addUnit($firstUnit);
-        $this->addUnit($secondUnit);
-
-        $this->getSubtotal()->shouldReturn(19000);
+        $firstUnit = new OrderItemUnit($this->getWrappedObject());
+        $adjustment1 = new Adjustment();
+        $adjustment1->setType(AdjustmentInterface::ORDER_UNIT_PROMOTION_ADJUSTMENT);
+        $adjustment1->setAmount(-1667);
+        $firstUnit->addAdjustment($adjustment1);
+
+        $secondUnit = new OrderItemUnit($this->getWrappedObject());
+        $adjustment2 = new Adjustment();
+        $adjustment2->setType(AdjustmentInterface::ORDER_UNIT_PROMOTION_ADJUSTMENT);
+        $adjustment2->setAmount(-1667);
+        $secondUnit->addAdjustment($adjustment2);
+
+        $secondUnit = new OrderItemUnit($this->getWrappedObject());
+        $adjustment3 = new Adjustment();
+        $adjustment3->setType(AdjustmentInterface::ORDER_UNIT_PROMOTION_ADJUSTMENT);
```

Comment:
We should have at least one adjustment of different type in this test

Reply:
Why?

Reply:
If I will remove the type from here: 
<img width="991" alt="image" src="https://user-images.githubusercontent.com/6213903/170761062-4912361f-b51e-4a00-bfbb-701b6dcc396f.png">
test will still pass. We may not catch it with higher level tests

Reply:
ok